runtime/vm/createramclass.cpp

```diff
@@ -2081,6 +2083,17 @@ checkFlattenableFieldValueClasses(J9VMThread *currentThread, J9ClassLoader *clas
 					result = FALSE;
 					break;
 				}
+
+				/* A NullRestricted field must be in a value class with an
+				 * ImplicitCreation attribute. The attribute must have the ACC_DEFAULT flag set.
+				 */
```

Comment:
Can you add a comment saying that the same check for static field is done in class preparation ?